src/components/organisms/AdminVenueView/components/Spaces/Spaces.tsx

```diff
@@ -0,0 +1,246 @@
+import RoomIconMusicBar from "assets/icons/icon-room-musicbar.svg";
+import RoomIconBurnBarrel from "assets/icons/icon-room-burnbarrel.svg";
+import RoomIconArtPiece from "assets/icons/icon-room-artpiece.svg";
+import RoomIconExperience from "assets/icons/icon-room-experience.svg";
+import RoomIconExternalLink from "assets/icons/icon-room-externallink.svg";
+import RoomIconMap from "assets/icons/icon-room-map.svg";
+
+import "./Spaces.scss";
+
+interface VenueRooms {
+  text: string;
+  template?: VenueRoomTemplate;
+  icon: string;
+}
+
+const venueRooms: VenueRooms[] = [
+  {
+    text: "Conversation Space",
+    icon: RoomIconConversation,
+    template: VenueTemplate.conversationspace,
+  },
+  {
+    text: "Auditorium",
+    icon: RoomIconAuditorium,
+    template: VenueTemplate.audience,
+  },
+  {
+    text: "Music Bar",
+    icon: RoomIconMusicBar,
+    template: VenueTemplate.jazzbar,
+  },
+  {
+    text: "Burn Barrel",
+    icon: RoomIconBurnBarrel,
+    template: VenueTemplate.firebarrel,
+  },
+  {
+    text: "Art Piece",
+    icon: RoomIconArtPiece,
+    template: VenueTemplate.artpiece,
+  },
+  {
+    text: "Experience",
+    icon: RoomIconExperience,
+    template: VenueTemplate.zoomroom,
+  },
+  {
+    text: "External link",
+    icon: RoomIconExternalLink,
+    template: RoomTemplate.external,
+  },
+  {
+    text: "Map",
+    icon: RoomIconMap,
+    template: VenueTemplate.partymap,
+  },
+];
+
+export interface SpacesProps {
+  venue: Venue_v2;
+  onClickNext: () => void;
+}
+
+export const Spaces: React.FC<SpacesProps> = ({ venue, onClickNext }) => {
+  const [selectedRoom, setSelectedRoom] = useState<RoomData_v2>();
+  const [showRooms, setShowRooms] = useState<boolean>(false);
+  const [updatedRoom, setUpdatedRoom] = useState<RoomData_v2>({});
+
+  const [showAddRoom, setShowAddRoom] = useState<boolean>(false);
+  const [showAdvancedSettings, setShowAdvancedSettings] = useState<boolean>(
+    false
+  );
+
+  const history = useHistory();
+
+  const hasSelectedRoom = !!selectedRoom;
+  const numberOfRooms = venue.rooms?.length ?? 0;
+
+  const clearSelectedRoom = useCallback(() => {
+    setSelectedRoom(undefined);
+    setUpdatedRoom({});
+  }, []);
+
+  const updateRoomPosition = useCallback(async (position: Position) => {
+    if (!position) return;
+
+    setUpdatedRoom((room) => ({
+      ...room,
+      x_percent: position.left,
+      y_percent: position.top,
+    }));
+  }, []);
+
+  const updateRoomSize = useCallback(async (size: Dimensions) => {
+    if (!size) return;
+
+    setUpdatedRoom((room) => ({
+      ...room,
+      width_percent: size.width,
+      height_percent: size.height,
+    }));
+  }, []);
+
+  const selectedRoomIndex =
+    venue.rooms?.findIndex((room) => room === selectedRoom) ?? -1;
+
+  return (
+    <div className="Spaces">
+      <div className="Spaces__rooms">
+        {hasSelectedRoom ? (
+          <EditRoomForm
+            room={selectedRoom!}
```

Comment:
I don't think it's a good approach. Something like `hasSelectedRoom && selectedRoom` should look better

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1583/commits/f8fa0c67c34ac626919737b4bbd112b4bd715553